crates/ide/src/typing.rs

```diff
@@ -373,4 +398,39 @@ fn main() {
     fn adds_space_after_return_type() {
         type_char('>', "fn foo() -$0{ 92 }", "fn foo() -> { 92 }")
     }
+
+    #[test]
+    fn adds_closing_brace() {
+        type_char('{', r"fn f() { match () { _ => $0() } }", r"fn f() { match () { _ => {()} } }");
+        type_char('{', r"fn f() { $0(); }", r"fn f() { {()}; }");
+        type_char('{', r"fn f() { let x = $0(); }", r"fn f() { let x = {()}; }");
+        type_char(
+            '{',
+            r"
```

Comment:
Let’s use standard  test style, with hashes and without indentation:

https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#minimal-tests